src/weather/weather.service.ts

```diff
@@ -0,0 +1,57 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { API } from '../constants/api-urls';
+import { ConfigService } from '@nestjs/config';
+import { AxiosResponse } from 'axios';
+import { firstValueFrom } from 'rxjs';
+import { WEATHER_ERROR } from '../errors/weather.errors';
+
+@Injectable()
+export class WeatherService {
+  private readonly apiKey: string;
+
+  constructor(
+    private readonly httpService: HttpService,
+    private readonly configService: ConfigService,
+  ) {
+    this.apiKey = this.configService.get<string>('WEATHER_API_KEY');
+  }
+
+  public async getForecast(
+    lat: number,
+    lng: number,
+  ): Promise<WeatherApiResponse> {
+    if (lat === undefined || lat === null || isNaN(lat)) {
+      throw new BadRequestException(WEATHER_ERROR.INVALID_LATITUDE_ERROR);
+    }
+    if (lng === undefined || lng === null || isNaN(lng)) {
+      throw new BadRequestException(WEATHER_ERROR.INVALID_LONGITUDE_ERROR);
+    }
```

Comment:
This won't be necessary when the query validation is done with `class-validator`, check the comment above.